### DISCUSSION TITLE
Audit history,  RBAC support, Change Requests

### DISCUSSION THREAD

**Discussion by achughZscaler** [flipt-io/flipt#1487] [General]
Any plans to support Audit history, Rbac and Change requests ( approve review process)?   Let me know if any of these are already supported and I missed them in Doc

**Comment by markphelps** [comment]
Hey @achughZscaler !

Audit history is coming in the next feature release version (likely v1.21.0)! 🎉 We're very close to having it completed, see https://github.com/flipt-io/flipt/discussions/1411 and https://github.com/flipt-io/flipt/issues/1469 for more. If you'd be up for testing a release-candidate version we'd love to chat with you!

We're currently writing all auditable events to a file that you specify via config, but will likely also support additional ways of consuming the audit events. Is there anything specifically you had in mind?

Re: Change requests, we are working on something to make change requests/reviews possible. I don't want to share too much as it's still very early and in the design phase, but would love to get your input on change approval and how you / your organization would like it to work! 

Finally RBAC, this isn't something we are actively working on but definitely something on our radar. Do you have existing tools/apps that you think do RBAC well? 

Thank you for the requests!

**Reply by achughZscaler** [reply]
Hey,

Thanks for quick response.  

For Audits - Mainly looking for what changed and who changed (if possible) to be displayed in Audit tab in the Ui. Storage can just be choice of DB. 

Change Request and RBAC maybe tied together - Use case would be a developer role can only add flag in an dev env or a specific namespace and then must be approved by QA role move to  prod or namespace. 

RBAC -  Limited access to see or add flags in specific namespaces.  Special permissions for delete etc etc.  

As for example - LaunchDarkly /Flagsmith   etc but they are upsells. 

I am just exploring tools right now and FLIPT seems to be becoming my fav to start these developments ( likely contribute here) that I need.

**Reply by markphelps** [reply]
@achughZscaler while we don't currently support storing the audit events to a DB/showing in the UI. We did just release log file based audit events in the [latest release](https://github.com/flipt-io/flipt/releases/tag/v1.21.0).

Would this work for you for the time being?

**Reply by achughZscaler** [reply]
Going to check it soon, thank you!

**Comment by GeorgeMac** [comment]
Just chiming in here to say that with the new git backend type (experimental in `v1.23.0`+) for Flipt you get audit history (git log), RBAC (codeowners) and change review (pull requests) for free. All in a GitOps package.
